AgentInterfaces/SessionHostsStartInfo.cs

```diff
@@ -167,21 +167,27 @@ public class MonitoringApplicationConfiguration
         /// <summary>
         /// Reference to the assets that contain the monitoring bits, to be used by the Performance profilers
         /// </summary>
-        public AssetDetail MonitoringAssetReference { get; set; }
+        public AssetDetail AssetReference { get; set; }
 
         /// <summary>
         /// Monitoring application installation script. 
         /// </summary>
-        public string MonitortingInstallationScript { get; set; }
+        public string InstallationScript { get; set; }
 
         /// <summary>
         /// Monitoring application running script. 
         /// </summary>
-        public string MonitoringExecutionScript { get; set; }
+        public string ExecutionScript { get; set; }
 
         /// <summary>
         /// Indicates whether the monitoring application should be run in on demand mode or from the beginning 
         /// </summary>
-        public bool MonitoringApplicationRunOnDemand { get; set; }
+        public bool RunOnDemand { get; set; }
+
+        /// <summary>
+        /// If the monitoring application is to be run from the beggining, for how long should it run.
```

Comment:
is this defined in hours/minutes/seconds? Could it be a Timespan?

Reply:
Also, how is it going to be triggered if not run from the beginning?

Reply:
Fixed the name, I want -1 to mean no limit, and TimeSpan would only have 0 which is a bit less clear IMO.

If the thing is not run from the beginning it will run until the caller server ends, or will be timebount in the GSDK call.